packages/http/libraries/core/src/http/calculations/requestParamFactory.ts

```diff
@@ -0,0 +1,38 @@
+import { Newable } from 'inversify';
+
+import { ControllerMethodParameterMetadata } from '../../routerExplorer/model/ControllerMethodParameterMetadata';
+import { CustomParameterDecoratorHandler } from '../models/CustomParameterDecoratorHandler';
+import { RequestMethodParameterType } from '../models/RequestMethodParameterType';
+import { Pipe } from '../pipe/model/Pipe';
+import { requestParam } from './requestParam';
+
+export function requestParamFactory(
+  parameterType: RequestMethodParameterType,
+  parameterPipeList: (Newable<Pipe> | Pipe)[],
+  parameterNameOrPipe?: string | (Newable<Pipe> | Pipe),
+  customParameterDecoratorHandler?: CustomParameterDecoratorHandler,
```

Comment:
```suggestion
  parameterName: string | undefined,
  customParameterDecoratorHandler: CustomParameterDecoratorHandler | undefined
```
Let's simplify the method signature

Reply:
This way we remove some funcionality. This function is supposed to abstract this complexity

Reply:
I think this method signature makes the code harder to read. I understand this signature in decorators such as `body`, but I would prefer to have simpler arguments here: one parameterName and a list of pipes. The duplicated logic can be avoided implementing some utils function. If you want I can go for it

Reply:
I understand your point, but this function has been created with the purpose of being that util function which has the responsibility to avoid such complexity; it is not a matter of avoiding certain logic. If the contract is configured in the proposed way, the decorators will always have to use a parameterName to be able to use pipes, and that is not the desired behavior.

Reply:
> I understand your point, but this function has been created with the purpose of being that util function which has the responsibility to avoid such complexity; it is not a matter of avoiding certain logic

I see. Not my cup of tea, but let's keep it that way then :slightly_smiling_face:.

> decorators will always have to use a parameterName to be able to use pipes, and that is not the desired behavior

Not really, parameterName will be `undefined` in that case.